### PR TITLE
[CDAP-13110] Fix dataset class loading in some places

### PR DESCRIPTION
**Pull request by anew**
Makes sure datasets are used and closed before the class loader goes out of scope.

### CODE REVIEW

**Comment by anew** [comment]
The dataset admin service needs more refactoring, but for now this should fix the issues I am seeing.

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminService.java [hunk 1]
```diff
@@ -223,37 +225,58 @@ public Void call() throws Exception {
     metadataStore.removeMetadata(datasetInstanceId);
   }
 
+  public boolean exists(DatasetId datasetInstanceId) throws Exception {
+    return performDatasetAdmin(datasetInstanceId, new Operation<Boolean>() {
```
Comment (chtyim) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminService.java]: Using method reference would be cleaner:

```
return performDatasetAdmin(datasetInstanceId, DatasetAdmin::exists);
```

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminService.java [hunk 2]
```diff
@@ -223,37 +225,58 @@ public Void call() throws Exception {
...
+        return datasetAdmin.exists();
+      }
+    });
+  }
+
   public void truncate(DatasetId datasetInstanceId) throws Exception {
-    DatasetAdmin datasetAdmin = getDatasetAdmin(datasetInstanceId);
-    datasetAdmin.truncate();
+    LOG.info("Truncating dataset {}", datasetInstanceId);
+    performDatasetAdmin(datasetInstanceId, new Operation<Void>() {
```
Comment (chtyim) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminService.java]: Same as above.

**Review by chtyim — COMMENTED** [review]
Just comment about using method references to have the code cleaner. The logical changes LGTM.

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminService.java (hunk 1 above)
Comment (anew) [reply cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminService.java]: Oh right we only support Java8 now. Good point.

**Comment by chtyim** [comment]
LGTM

**Comment by chtyim** [comment]
LGTM

**Comment by anew** [comment]
build passed: https://builds.cask.co/browse/CDAP-DUT6289-6

I will now squash and merge.

**Pull request merged**